### DISCUSSION TITLE
Surface 3 wifi issue — please confirm solution

### DISCUSSION THREAD

**Discussion by writersglen** [linux-surface/linux-surface#485] [Q&A]
Hello,

I’ve installed Ubuntu 20.04 on an MS Surface 3. I’m experiencing the widely reported wifi dropout problem. Extensive Google research has led me to believe that I can solve the problem by replacing the Linux kernel with the most recent linux-surface kernel. Is this correct?

uname -r shows: 5.8.0-59-generic

Many thanks,

LRP

**Comment by qzed** [comment]
Yes, the Surface kernels have some patches that should improve WiFi. I can't speak for the Surface 3, but WiFi stability is significantly improved by those patches on my Surface Book 2.

**Comment by writersglen** [comment]
Thank you Maxmilian!

LRP

**Comment by writersglen** [comment]
I’m setting out to upgrade the kernel on my MS Surface 3 to the linux-surface kernel following these instructions:

https://github.com/linux-surface/linux-surface/wiki/Installation-and-Setup

The post installation section suggests that wifi instability may still persist. Can I expect this? Or have these issues been resolved in the very latest kernel release?  

My fear is falling ever deeper down into a rabbit hole.

Thanks,

LRP

**Reply by qzed** [reply]
Ah that was a bit outdated. The tweaks suggested there shouldn't be required any more. I can't give you any guarantees (since I don't own a S3) but things should be quite a bit better than on the stock kernel.

**Comment by writersglen** [comment]
Hi Maxmilian,

Just tried the update. Big pain, copy/paste doesn’t seem to work so I arduously hand typed the long lines. Perhaps made a mistake somewhere since the install command tells me that it can’t locate any of the packages.

Must be a better way. May try again when I less hot and tired. 

Thanks,

LRP
